book/rails/introduction.md

```diff
@@ -1,181 +1,186 @@
-combination required for all API requests. Requiring a client id and secret
-ensures that only known users can access the API. This also allows the API to
-turn off access to a particular person or application if the usage violates the
-API terms of service. [This blog post on API
-security](https://stormpath.com/blog/top-six-reasons-use-api-keys-and-how/)
-explains the benefits of API keys.
-
-Humon is not an API built for commercial purposes, so we don't need to worry
-about creating a complicated permissions scheme that involves many API tokens.
-We do, however, want to make sure that not just anyone can query our API. Having
-zero security would mean that anyone could create a `curl` request to an
-endpoint and hit our database.
-
-As a security measure, we require a header of `tb-app-secret` for the POST users
-request. We ensure that only requests that send a `tb-app-secret` header with a
-value matching the value set by the Rails app can create new users via the Humon
-API. We store the app secret in an environment variable so that it is not
-anywhere in version control.
-
-We use [`dotenv`](https://github.com/bkeepers/dotenv) so that our API can read
-environment variables from a `.env` file while in development mode. We also add
-`.env` to `.gitignore` so it is not added to our git repository. If the app
-secret sent by the client on POST users does not match the app secret in the
-API, a `404 Not Found` is returned.
-
-For all requests (except for a POST users request), we require that the header
-contain a `tb-auth-token`. During a POST users request, we create an auth token
-for a user and return it in the response JSON. The iOS app stores
-that token and sets it in the header of every subsequent request.
-
-To see how we implemented an app secret and auth tokens for POST users, see the
-[`before_filter` in our `UsersController`](https://github.com/thoughtbot/ios-on-rails/blob/master/example_apps/rails/app/controllers/api/v1/users_controller.rb)
-and the [`before_validation` in our `User model`.](https://github.com/thoughtbot/ios-on-rails/blob/master/example_apps/rails/app/models/user.rb)
+client secret. These strings are unguessable. They act as a username and
+password combination required for all API requests. Requiring a client id and
+secret ensures that only known users can access the API. This also allows the
+API to turn off access to a particular person or application. If their usage
+violates the API terms of service, we can disable their key. [This blog post on
+API security][] explains the benefits of API keys.
+
+Humon is an API built for non-commercial purposes. We can create a simple
+permission scheme without many API tokens. But, we want to ensure that only some
+people can query our API. Having zero security would mean anyone could create a
+`curl` request to an endpoint. They could hit our database and cause unknown
+damage.
+
+As a security measure, we need a header of `tb-app-secret` for the `POST users`
```

Comment:
What do you think of removing the `tb-` prefix, as this is not a thoughtbot-global parameter? Or changing it to something linked to Humon and not thoughtbot?